pkg/engine/imageVerify.go

```diff
@@ -219,27 +217,27 @@ func (iv *imageVerifier) verifySignature(imageVerify *v1.ImageVerification, imag
 	return ruleResp, digest
 }
 
-func (iv *imageVerifier) patchDigest(imageInfo context.ImageInfo, digest string, ruleResp *response.RuleResponse) {
+func (iv *imageVerifier) patchDigest(path string, imageInfo imageutils.ImageInfo, digest string, ruleResp *response.RuleResponse) {
 	if imageInfo.Digest == "" {
-		patch, err := makeAddDigestPatch(imageInfo, digest)
+		patch, err := makeAddDigestPatch(path, imageInfo, digest)
 		if err != nil {
-			iv.logger.Error(err, "failed to patch image with digest", "image", imageInfo.String(), "jsonPath", imageInfo.JSONPointer)
+			iv.logger.Error(err, "failed to patch image with digest", "image", imageInfo.String(), "jsonPath", path)
 		} else {
 			iv.logger.V(4).Info("patching verified image with digest", "patch", string(patch))
 			ruleResp.Patches = [][]byte{patch}
 		}
 	}
 }
 
-func makeAddDigestPatch(imageInfo context.ImageInfo, digest string) ([]byte, error) {
+func makeAddDigestPatch(path string, imageInfo imageutils.ImageInfo, digest string) ([]byte, error) {
 	var patch = make(map[string]interface{})
 	patch["op"] = "replace"
-	patch["path"] = imageInfo.JSONPointer
+	patch["path"] = path
 	patch["value"] = imageInfo.String() + "@" + digest
```

Comment:
Although not introduced in this PR, this logic is also incorrect as it leads to final images like - 

`ghcr.io/kyverno/kyverno:latest@sha256:...`

instead of `ghcr.io/kyverno/kyverno@sha256:...`

Reply:
Is that incorrect, though? I recall reading somewhere that the tag is allowed with the digest for readability, but not used when the digest is specified. Are there cases when it would cause a problem if the tag is specified?

Reply:
Some tools support it and some don't (depending on the version). For eg. https://github.com/cri-o/cri-o/issues/2351#issuecomment-583158833

I would prefer it if kyverno didn't inject this by default.